### PR TITLE
[JENKINS-54964] Add debugging information to the documentation

### PR DESCRIPTION
**Pull request by MRamonLeon**
See [JENKINS-54964](https://issues.jenkins-ci.org/browse/JENKINS-54964), this task has been created during the java11 support epic, but it has nothing to do with java11.

Add some info about how to debug using the docker container.

@reviewbybees @jenkinsci/java11-support @olivergondza

### CODE REVIEW

File: docs/DOCKER.md [hunk 1]
```diff
@@ -11,3 +11,39 @@ Interactive shell:
...
+
+Launch the container:
+
+    mramonleon@laptop:~/acceptance-test-harness$ ./ath-container.sh
+
+In the container shell, run the tests with debugging :
+
+    ath-user@1803848e337f:~/ath-sources$ ./run.sh firefox latest -Dmaven.surefire.debug="-Xdebug -Xrunjdwp:transport=dt_socket,server=y,suspend=y,address=*:5005 -Xnoagent -Djava.compiler=NONE" -DrunSmokeTests`
+
+In other terminal of the host, not in the container, get the name and IP of the container used:
```
Comment (jeffret-b) [inline docs/DOCKER.md]: I suggest "In other terminal of the host" -> "In another terminal on the host".

File: docs/DOCKER.md [hunk 2]
```diff
@@ -11,3 +11,39 @@ Interactive shell:
 Jenkinsfile:
 
 See the repository `Jenkinsfile` for inspiration.
+
+# Debugging tests in a docker container
+If you need to debug the tests running in a docker container, you need to run the tests inside the container with:
+
+    ./run.sh firefox latest -Dmaven.surefire.debug="-Xdebug -Xrunjdwp:transport=dt_socket,server=y,suspend=y,address=*:5005 -Xnoagent -Djava.compiler=NONE" -Dtest=...`
+
+With this instruction, the tests will stop until you connect a remote debugger. You can do that connecting your IDE to the **IP** used by the container in the port **5005**. You can figure out the IP used by the container with:
```
[the remark this answers is not in the document]
Comment (MRamonLeon) [reply docs/DOCKER.md]: modified using the first suggestion

File: docs/DOCKER.md [hunk 3]
```diff
@@ -11,3 +11,39 @@ Interactive shell:
...
+
+    mramonleon@laptop:~/acceptance-test-harness$ ./ath-container.sh
+
+In the container shell, run the tests with debugging :
+
+    ath-user@1803848e337f:~/ath-sources$ ./run.sh firefox latest -Dmaven.surefire.debug="-Xdebug -Xrunjdwp:transport=dt_socket,server=y,suspend=y,address=*:5005 -Xnoagent -Djava.compiler=NONE" -DrunSmokeTests`
+
+In other terminal of the host, not in the container, get the name and IP of the container used:
+
+    mramonleon@laptop:~/acceptance-test-harness$ docker ps
```
[the remark this answers is not in the document]
Comment (MRamonLeon) [reply docs/DOCKER.md]: I used this to keep it clear when it's run in the host, or in the container. I can use _harry@devbox_ as before, I didn't realize it is a ficticious user/host.

File: docs/DOCKER.md [hunk 4]
```diff
@@ -11,3 +11,39 @@ Interactive shell:
...
+
+With this instruction, the tests will stop until you connect a remote debugger. You can do that connecting your IDE to the **IP** used by the container in the port **5005**. You can figure out the IP used by the container with:
+
+    docker inspect -f '{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}' NAME_OF_THE_CONTAINER_RUNNING
+
+TIP: to know the name of the container execute `docker ps`
+
+It's best explained with an example:
+
+Launch the container:
```
[the remark this answers is not in the document]
Comment (MRamonLeon) [reply docs/DOCKER.md]: changed

File: docs/DOCKER.md [hunk 5]
```diff
@@ -11,3 +11,39 @@ Interactive shell:
...
+
+    mramonleon@laptop:~/acceptance-test-harness$ ./ath-container.sh
+
+In the container shell, run the tests with debugging :
```
[the remark this answers is not in the document]
Comment (MRamonLeon) [reply docs/DOCKER.md]: fixed

File: docs/DOCKER.md (hunk 1 above)
Comment (MRamonLeon) [reply docs/DOCKER.md]: changed.

**Comment by MRamonLeon** [comment]
Last version, after @jeffret-b comments: https://github.com/jenkinsci/acceptance-test-harness/blob/87618044871fe26b22c622c6de7383815ec76ea5/docs/DOCKER.md

**Comment by MRamonLeon** [comment]
Last version, after @jeffret-b comments:https://github.com/jenkinsci/acceptance-test-harness/blob/0b976f6da98c83d47550dbb83574fa9988a20c7c/docs/DOCKER.md

File: docs/DOCKER.md [hunk 6]
```diff
@@ -3,11 +3,57 @@
...
+docker inspect -f '{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}' NAME_OF_THE_CONTAINER_RUNNING
+```
+
+*TIP: to know the name of the container execute* `docker ps`
+
+It's best explained with an example:
+
+1. Launch the container:
+   ```bash
+   harry@devbox:~/acceptance-test-harness$ ./ath-container.sh
```
[the remark this answers is not in the document]
Comment (MRamonLeon) [reply docs/DOCKER.md]: Does it sound good to you if I add the _vnc.sh_ step before @olivergondza?

**Comment by MRamonLeon** [comment]
last version: https://github.com/jenkinsci/acceptance-test-harness/blob/00360349a38a0505dc287105e0537b11cd6512fe/docs/DOCKER.md

**Comment by olivergondza** [comment]
Thanks, merged!

**Pull request merged**